scripts/testing/cli.py

```diff
@@ -0,0 +1,603 @@
+        connector_command = [
+            "connector",
+            "create",
+            "--index-name",
+            scenario["index_name"],
+            "--service-type",
+            scenario["service_type"],
+            "--index-language",
+            scenario["index_language"],
+            native,
+            "--from-file",
+            os.path.basename(scenario["connector_configuration"]),
+            "--update-config",
+            "--connector-service-config",
+            CONNECTOR_SERVICE_CONFIG_FILE,
+        ]
+        cmd = [
+            "gcloud",
+            "compute",
+            "ssh",
+            name,
+            "--zone",
+            vm_zone,
+            "--command",
+            f"sudo /var/app/bin/connectors -c {CLI_CONFIG_FILE} {' '.join(connector_command)}",
+        ]
+
+        result = run_gcloud_cmd(cmd)
+        connector_id = re.search("ID:\s([\w\-_]+)", result.stdout.decode("utf-8"))[1]
+
+        # Start the service
+        cmd = [
+            "gcloud",
+            "compute",
+            "ssh",
+            name,
+            "--zone",
+            vm_zone,
+            "--command",
+            f"sudo /var/app/bin/elastic-ingest -c {CONNECTOR_SERVICE_CONFIG_FILE} --debug --filebeat >~/service.log 2>&1 &",
+        ]
+        run_gcloud_cmd(cmd)
+
+        for test in scenario["tests"]:
+            click.echo(f"Run test: {test['name']}")
+            cmd = [
+                "gcloud",
+                "compute",
+                "ssh",
+                name,
+                "--zone",
+                vm_zone,
+                "--command",
+                f"sudo /var/app/bin/connectors -c {CLI_CONFIG_FILE} job start -i {connector_id} -t {test['job_type']} --format json",
+            ]
+            result = run_gcloud_cmd(cmd)
+
+            job_id = json.loads(result.stdout.decode("utf-8"))["id"]
+
+            timeout = 0
+            timeout_step = test["max_timeout"] / 5
+            while True:
+                cmd = [
+                    "gcloud",
+                    "compute",
+                    "ssh",
+                    name,
+                    "--zone",
+                    vm_zone,
+                    "--command",
+                    f"sudo /var/app/bin/connectors -c {CLI_CONFIG_FILE} job view {job_id} --format json",
+                ]
+                result = run_gcloud_cmd(cmd)
+                job = json.loads(result.stdout.decode("utf-8"))
+
+                if job["job_status"] != test["match"]["status"]:
+                    time.sleep(timeout_step)
+                    timeout += timeout_step
+                else:
+                    click.echo(click.style(f"Test {test['name']} passed", fg="green"))
+                    break
+
+                if timeout >= test["max_timeout"]:
+                    click.echo(click.style(f"Test {test['name']} failed", fg="red"))
+                    break
+
+
+# TODO Read all the fields in one call
```

Comment:
Should we address this before merging, or is it okay for later?

How many keys do we expect?

Reply:
I would address it in a separate PR if there will be significant performance issues.  I think each test can have 5-7 calls which is okay. Timewise it's less than 1% of the entire test.